Infrastructure/Analytics/AnalyticsConfigurerService.cs

```diff
@@ -0,0 +1,74 @@
+﻿using Dfe.Analytics.AspNetCore;
+using Google.Apis.Auth.OAuth2;
+using Google.Cloud.BigQuery.V2;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+
+namespace UI.Analytics
+{
+    interface IAnalyticsConfigurerService
+    {
+        void ConfigureApp(IApplicationBuilder app);
+    }
+
+    public class AnalyticsConfigurerService : IAnalyticsConfigurerService
+    {
+        private readonly IConfiguration _configuration;
+        private readonly ILogger _logger;
+        private readonly bool _isConfigured;
+
+        public AnalyticsConfigurerService(IConfiguration configuration, ILogger<AnalyticsConfigurerService> logger)
+        {
+            this._configuration = configuration;
+            this._logger = logger;
+            this._isConfigured = IsAnalyticsConfigured();
+        }
+
+        public void ConfigureApp(IApplicationBuilder app)
+        {
+            if (_isConfigured)
+            {
+                app.UseDfeAnalytics();
+            }
+            else
+            {
+                _logger.LogWarning("DfeAnalytics is NOT configured and will be disabled");
+            }
+        }
+
+        private bool IsAnalyticsConfigured()
+        {
+            var section = _configuration.GetSection("DfeAnalytics");
+
+            // Do we have the bare minimum of configuration?
+            return (
+                (
+                    !string.IsNullOrEmpty(section["CredentialsJson"])
+                    || (!string.IsNullOrEmpty(section["CredentialsJsonFile"]) && !string.IsNullOrEmpty(section["ProjectId"]))
```

Comment:
Since using an OR, I'm not sure it will ever check the Project Id in this clause?

Reply:
I think the logic is correct, but I agree it's not very obvious what the logic is supposed to be doing! I'll tidy it up a bit.

Reply:
Have rearranged the logic a bit.